web/handlers/http.go

```diff
@@ -221,7 +202,14 @@ func New(
 	// hookup handlers to the router
 	roomsAuth.Handler(m, r, a)
 
-	adminHandler := a.Authenticate(admin.Handler(r, roomState, al, ns, ps))
+	adminHandler := a.Authenticate(admin.Handler(
+		domainName,
+		r,
+		roomState,
+		al,
+		is,
+		ns,
+		ps))
```

Comment:
would it be completely ludicrous to pass in a struct instead (and get some documentation, cause like lol) similar to

```go
	var nh = noticeHandler{
		notices: ns,
		pinned:  ps,
	}
```

Reply:
I hear you. I will try to come up with something more fitting.

I'm trying to not end up in a place where a struct with all the databases is used everywhere. This case is indeed a bit extreme.

Reply:
see d302423  and 78a8c3b